lighthouse-core/gather/drivers/driver.js

```diff
@@ -213,6 +221,76 @@ class Driver {
     });
   }
 
+  getServiceWorkerRegistrations() {
+    return new Promise(resolve => {
+      this.once('ServiceWorker.workerRegistrationUpdated', data => {
+        this.sendCommand('ServiceWorker.disable');
+        resolve(data);
+      });
+
+      this.sendCommand('ServiceWorker.enable');
+    });
+  }
+
+  /**
+   * Checks all serviceworkes and see if any duplications are running
+   * @param {!string} pageUrl
+   * @return {!Promise}
+   */
+  checkForMultipleTabsAttached(pageUrl) {
+    // Get necessary information of serviceWorkers
+    const getRegistrations = this.getServiceWorkerRegistrations();
+    const getVersions = this.getServiceWorkerVersions();
+    const getActiveTabId = this.getCurrentTabId();
+
+    return Promise.all([getRegistrations, getVersions, getActiveTabId]).then(res => {
+      const registrations = res[0].registrations;
+      const versions = res[1].versions;
+      const activeTabId = res[2];
+      const parsedURL = parseURL(pageUrl);
+      const origin = `${parsedURL.protocol}//${parsedURL.hostname}` +
+      (parsedURL.port ? `:${parsedURL.port}` : '');
+
+      let swHasMoreThanOneClient = false;
+      registrations
+        .filter(reg => !reg.isDeleted)
+        .filter(reg => {
+          const parsedURL = parseURL(reg.scopeURL);
+          const swOrigin = `${parsedURL.protocol}//${parsedURL.hostname}` +
+            (parsedURL.port ? `:${parsedURL.port}` : '');
+
+          return origin === swOrigin;
+        })
+        .forEach(reg => {
+          // Check if any of the service workers are the same (registration id)
+          // Check if the controlledClients are bigger than 1 and it's not the active tab
+          swHasMoreThanOneClient = !!versions.find(ver => {
+            if (ver.registrationId !== reg.registrationId) {
+              return false;
+            }
+
+            if (!ver.controlledClients || ver.controlledClients.length === 0) {
+              return false;
+            }
+
+            const multipleTabsAttached = ver.controlledClients.length > 1;
+            const oneInactiveTabIsAttached = ver.controlledClients.length === 1 &&
+              !ver.controlledClients.find(client => client === activeTabId);
+
+            return multipleTabsAttached || oneInactiveTabIsAttached;
+          });
+        });
+
+      if (swHasMoreThanOneClient) {
+        throw new Error(
```

Comment:
only needed to disconnect the driver. Couldn't find anything else. I will have a look at what happens when we close the trace in between.. (shouldn't be part of this PR)